hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionsEntityResourceTest.java

```diff
@@ -43,21 +44,24 @@
 import org.alfresco.hxi_connector.hxi_extension.rest.api.config.QuestionsApiConfig;
 import org.alfresco.hxi_connector.hxi_extension.rest.api.model.QuestionModel;
 import org.alfresco.hxi_connector.hxi_extension.service.HxInsightClient;
+import org.alfresco.hxi_connector.hxi_extension.service.QuestionPermissionService;
 import org.alfresco.hxi_connector.hxi_extension.service.model.RestrictionQuery;
 
 public class QuestionsEntityResourceTest
 {
     private static final String AGENT_ID = "agent-id";
 
     private final HxInsightClient hxInsightClient = mock(HxInsightClient.class);
-    private final QuestionsApiConfig questionConfig = new QuestionsApiConfig(100);
-    private final QuestionsEntityResource questionsEntityResource = new QuestionsEntityResource(hxInsightClient, questionConfig);
+    private final QuestionsApiConfig questionConfig = new QuestionsApiConfig(3);
+    private final QuestionPermissionService questionPermissionService = mock(QuestionPermissionService.class);
+    private final QuestionsEntityResource questionsEntityResource = new QuestionsEntityResource(hxInsightClient, questionConfig, questionPermissionService);
 
     @Test
     public void shouldFailIfAskedMultipleQuestions()
     {
         // given
         List<QuestionModel> questions = List.of(mock(QuestionModel.class), mock(QuestionModel.class));
+        given(questionPermissionService.hasPermissionToAskAboutDocuments(any())).willReturn(true);
```

Comment:
Maybe this line could be located in `@BeforeEach`? Then it won't be necessary in every test requiring happy case

Reply:
done